src/main/scala/org/graphframes/lib/ConnectedComponents.scala

```diff
@@ -356,12 +362,14 @@ object ConnectedComponents extends Logging {
 
       ee.persist(intermediateStorageLevel)
 
-      // test convergence
+      minNbrs1 = minNbrs(ee) // src >= min_nbr
+        .persist(intermediateStorageLevel)
```

Comment:
Within the termination of the while loop, should we also explicitly call unpersist() in order to remove the cache references to these pseudo-mutable vars that represent an immutable cache state?

Reply:
I will revisit and fix the persist/unpersist operations